File: algorithms/python/reverse-integer.py
def reverse(num):
    """

    :param num: int
    :return: int
    """
    sign = -1 if num < 0 else 1
    num = abs(num)
    ans = 0
    while num:
        # ans = ans * 10 + num % 10
        # num /= 10
        num, mod = divmod(num, 10)
        ans = ans * 10 + mod

    return sign * ans if ans <= 0x7fffffff else 0

File: algorithms/php/reverse-integer.php
<?php

/**
 * @param $num
 * @return int
 */
function reverseInteger($num)
{
    $sign = 1;
    $ans  = 0;
    if ($num < 0) {
        $sign = -1;
        $num  *= -1;
    }

    while ($num > 0) {
        $tmp = $num % 10;
        $ans = $ans * 10 + $tmp;

        $num = intval($num / 10);
    }

    $ans *= $sign;

    if ($ans > PHP_INT_MAX || $ans < PHP_INT_MIN) {
        return 0;
    }
    return $ans;
}